### PR TITLE
Religion Systems & Mechanics

### PR DESCRIPTION
**Pull request by ZeroDayDaemon**
## About the PR

Implements several features revolving around religion.

- [x] Way to select religion on character setup.
- [x] Atheists cannot pray.
- [x] Chaplain spawns with a unique bible based on their faith
- [x] When Non-atheist chaplain uses bible on atheist, set them on fire.
- [x] When Atheist chaplain uses bible on non-atheist, set them on fire.

## Why / Balance

New and interesting gameplay mechanics, interactions, and features for chaplain and crew.
## Technical details

Religion Component is mostly just a tag with some fields for other systems to check.
Edited some systems to change how they interact with entities with the Religion component.
People without religion set will be treated as the same religion as the person using the bible on them

**Changelog**

:cl: Daemon
- add: Added religions.
- add: Added new types of bibles.

## Summary by CodeRabbit

- **New Features**
	- Introduced a `ReligionComponent` to facilitate religion-based interactions, affecting gameplay mechanics.
	- Added conditional interactions based on users' religious affiliations, including fire effects and notifications.
	- Enhanced prayer functionality with notifications for users identified as atheists.
	- Introduced new religious traits (Atheist, Buddhist, Christian) for character customization.
	- Added new bible items (`goobbit`, `tripitaka`) to enrich gameplay lore.

- **Localization Updates**
	- Added new strings for prayer notifications and interactions to enhance user engagement and inclusivity.
	- Updated localization for religious traits and the profile editor to reflect new features.

### CODE REVIEW

**Comment by ZeroDayDaemon** [comment]
This PR is in draft while I work on all the features and get feedback!

**Comment by cheki-ss14** [comment]
I will heavily abuse this 👍

**Comment by Verbalase** [comment]
No. I do NOT want this, bro.

**Comment by BombasterDS** [comment]
Please no

**Comment by JoeHammad1844** [comment]
if you fart on a bible you should gib

**Comment by JoeHammad1844** [comment]
i support this 100%

**Comment by lanse12** [comment]
close this please

**Comment by lexs0sa** [comment]
I wanna fart in my dad's mouth

**Comment by lexs0sa** [comment]
> i support this 100%

I wanna fart in your mouth

**Comment by lexs0sa** [comment]
oh my god this would make me so "effing" horny oh my God please PR merge this, GOD this needs upstream NOW. Bo Sinn's trenchcoat flies off

**Comment by JoeHammad1844** [comment]
> I wanna fart in your mouth

what did he mean by this?

**Review by whateverusername0 — APPROVED** [review]
i approve this already

**Comment by Piras314** [comment]
Religion changes need a lot of work.

Chaplains should be able to pick religions, one of them being "Atheist" despite it not being a religion, and the book they'll be given is Reddit.

The Reddit book will deal damage to non-atheists and heal atheists, etc, etc.

Atheists should also be able to open Christmas presents unless there's some alternative being offered.

**Comment by ZeroDayDaemon** [comment]
> Atheists should also be able to open Christmas presents unless there's some alternative being offered.
They can still open them, they just get shocked if they aren't insulated

**Comment by ZeroDayDaemon** [comment]
I'll look into moving religion into an Enum

**Comment by ZeroDayDaemon** [comment]
I'm pulling back the scope of this PR to be just religion related content. Farting will be made into it's own PR

**Comment by ZeroDayDaemon** [comment]
@coderabbitai review

**Comment by ZeroDayDaemon** [comment]
> I'm sorry, but I can't assist with that.

My PR too spicy

File: Content.Server/Holiday/Christmas/RandomGiftSystem.cs [hunk 1]
```diff
@@ -57,6 +65,13 @@ private void OnUseInHand(EntityUid uid, RandomGiftComponent component, UseInHand
         if (component.SelectedEntity is null)
             return;
 
+        //Public Domain Code start
+        if (EntityManager.TryGetComponent(args.User, out ReligionComponent? religionComponent) && religionComponent.Type == Religion.Atheist)
+        {
+            _electrocutionSystem.TryDoElectrocution(args.User, uid, 25, TimeSpan.FromSeconds(5), true);
+        }
```
[the remark this answers is not in the document]
Comment (ZeroDayDaemon) [reply Content.Server/Holiday/Christmas/RandomGiftSystem.cs:72]: As funny as this is, the reception has been negative, and it's already an incredibly niche interaction. removing it.

File: Content.Server/Bible/BibleSystem.cs [hunk 2]
```diff
@@ -115,6 +120,53 @@ private void OnAfterInteract(EntityUid uid, BibleComponent component, AfterInter
...
+                                _flammableSystem.SetFireStacks(args.Target.Value, 1);
+                                _flammableSystem.Ignite(args.Target.Value, args.User);
+                                _delay.TryResetDelay((uid, useDelay));
+                                return;
+                            }
+                            return;
+                        }
+                        break;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (ZeroDayDaemon) [reply Content.Server/Bible/BibleSystem.cs]: Reduced code duplication

**Comment by BombasterDS** [comment]
me when farting removed 🥰

**Comment by JoeHammad1844** [comment]
farting removed? i am mortified

**Comment by ZeroDayDaemon** [comment]
Farting moved, not REmoved, will be another PR

**Comment by ZeroDayDaemon** [comment]
@coderabbitai review

**Comment by BombasterDS** [comment]
> Farting moved, not REmoved, will be another PR

please no

File: Content.Server/Bible/BibleSystem.cs [hunk 3]
```diff
@@ -155,6 +186,24 @@ private void OnAfterInteract(EntityUid uid, BibleComponent component, AfterInter
             }
         }
 
+        //Public Domain Code begin
```
Comment (whateverusername0) [inline Content.Server/Bible/BibleSystem.cs:189]: can you change all public domain code comments into goob edits?
like `//goob edit - religion`

File: Content.Server/Goobstation/Religion/ReligionSystem.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            if (EntityManager.TryGetComponent(args.Mob, out ReligionComponent? mobReligion))
+            {
+                var bible = mobReligion.Type switch
+                {
+                    Shared.Goobstation.Religion.Religion.Atheist => "BibleAtheist",
+                    Shared.Goobstation.Religion.Religion.Buddhist => "BibleBuddhist",
+                    Shared.Goobstation.Religion.Religion.Christian => "Bible",
+                    Shared.Goobstation.Religion.Religion.None => "Bible",
+                };
+                //_inventorySystem.SpawnItemOnEntity(args.Mob, bible); Why doesnt this have a silent option ffs
```
Comment (whateverusername0) [inline Content.Server/Goobstation/Religion/ReligionSystem.cs:33]: why is it here and why is it unused

File: Content.Shared/Goobstation/Religion/ReligionComponent.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+public sealed partial class ReligionComponent : Component
+{
+    /// <summary>
+    ///     The religion of the entity
+    /// </summary>
+    [DataField("religion")] [AutoNetworkedField]
+    public Religion Type = Religion.None;
+}
+
+public enum Religion
```
Comment (whateverusername0) [inline Content.Shared/Goobstation/Religion/ReligionComponent.cs:21]: no muslimeen? :trollface:

**Review by whateverusername0 — CHANGES_REQUESTED** [review]
looks good overall but i have some questions

File: Content.Shared/Goobstation/Religion/ReligionComponent.cs (hunk 5 above)
Comment (ZeroDayDaemon) [reply Content.Shared/Goobstation/Religion/ReligionComponent.cs:21]: Not touching that one, someone else can add it

File: Content.Server/Goobstation/Religion/ReligionSystem.cs (hunk 4 above)
Comment (ZeroDayDaemon) [reply Content.Server/Goobstation/Religion/ReligionSystem.cs:33]: Testing and bitching initially, probably forgot it

File: Content.Server/Bible/BibleSystem.cs (hunk 3 above)
Comment (ZeroDayDaemon) [reply Content.Server/Bible/BibleSystem.cs:189]: I can add that to it yes, but it will still retain the public domain code comment as well

**Comment by ZeroDayDaemon** [comment]
@coderabbitai review

**Review by whateverusername0 — APPROVED** [review]
looks good

**Pull request merged**